contract_api/consumers/service_event_consumer.py

```diff
@@ -112,6 +112,62 @@ def _get_new_assets_url(self, org_id, service_id, new_ipfs_data):
                                                                  service_id)
         return assets_url_mapping
 
+    def create_service_media(self,org_id,service_id,service_media,service_row_id,assets_url):
+        count = 0;
+        if(service_media!=None):
+            #get existing media data
+            existing_service_media_data = self._service_repository.get_service_media(service_id=service_id,org_id=org_id)
+
+            #delete existing media files if present form s3 bucket except video which is not saved
+            if existing_service_media_data!=None:
+                for media in existing_service_media_data:
+                    if(len(media['url'])>0 and media('file_type')!="video"):
+                        self._s3_util.delete_file_from_s3(url=media('url'))
+
+            #clear the existing values from db
+            self._service_repository.delete_service_media(org_id=org_id,service_id=service_id)
+
+            #for media other that video save to s3 and store the link details in db
+            for service_media_item in service_media:
+                if service_media_item["file_type"] == "video":
```

Comment:
What if somebody has uploaded a mp4 file? Not all video files need to be links correct? In fact somebody may have provided a link for an image as well

Reply:
This was discussed with Anand and for now we are giving url back for type videos

Reply:
we need to handle this : so if the url is an ipfs link => we need to download and store it in the S3 , if its a link  then use it as is for videos

Reply:
This is not restricted or even related to videos. Any asset that has an ipfs link must be uploaded to S3 to serve it. It could be a video or an image

Reply:
this change is updated

Reply:
this is updated for http/https and url links